### PR TITLE
Create prepare-host github actions

### PR DESCRIPTION
**Pull request by mangelajo**

### CODE REVIEW

File: .github/workflows/consuming.yml [hunk 1]
```diff
@@ -39,17 +39,9 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Install WireGuard kernel module (only in WireGuard jobs)
```
Comment (mkolesnik) [inline .github/workflows/consuming.yml:41]: Should probably change the description :)

File: gh-actions/prepare-host/action.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: 'Prepare host'
```
Comment (mkolesnik) [inline gh-actions/prepare-host/action.yaml:1]: Generally this is only ever used for E2E, can we just fold it into the "Run E2E" shared action?

File: test/e2e/framework/network_pods.go [hunk 3]
```diff
@@ -282,7 +282,7 @@ func (np *NetworkPod) buildThroughputClientPod() {
 					Name:            "nettest-client-pod",
```
Comment (mkolesnik) [inline test/e2e/framework/network_pods.go:282]: Why was this file changed?

**Review by mkolesnik — CHANGES_REQUESTED** [review]
Interesting :)

File: test/e2e/framework/network_pods.go (hunk 3 above)
Comment (mangelajo) [reply test/e2e/framework/network_pods.go:282]: PEBKAK

File: gh-actions/prepare-host/action.yaml (hunk 2 above)
Comment (mangelajo) [reply gh-actions/prepare-host/action.yaml:1]: What do you mean, wait...

Comment (mkolesnik) [reply gh-actions/prepare-host/action.yaml:1]: We only need to "prepare host" before we run the E2E.
I think this can all be in a shared "run E2E" shared action (and if necessary we can split it out in the future though I doubt it will be).
